snooty/parser.py

```diff
@@ -917,6 +920,15 @@ def validate_list_table(
             )
             return
 
+    def validate_tabs_children(self, node: n.Directive) -> None:
+        line = node.start[0]
+        for child in node.children:
+            if (not isinstance(child, n.Directive)) or child.name != "tab":
+                self.diagnostics.append(
+                    TabMustBeDirective(str(type(child).__class__.__name__), line)
+                )
+        return
```

Comment:
This essentially replicates the existing logic we had for validating child nodes in `handle_tabset`. This could be DRYer, but I'm not sure if there's a natural place in the repo to consolidate that logic.